src/_pytest/cacheprovider.py

```diff
@@ -85,6 +85,13 @@ def set(self, key, value):
                python types, including nested types
                like e. g. lists of dictionaries.
         """
+        content_readme = 'test'
+        if os.path.isdir(self._cachedir):
+            readme_path = os.path.join(self._cachedir, 'README.md')
+            if os.path.isfile(readme_path) is False:
```

Comment:
The idiomatic way is to use `if not os.path.isfile(readme_path)`. Also, it is probably easier to use the [`py.path` API](http://py.readthedocs.io/en/latest/path.html) directly:

```python
        if self._cachedir.check(dir=True):
            readme_path = self._cachedir.join('README.md')
            if not readme_path.check(file=True):
                readme_path.write(content_readme)
```